### PR TITLE
commit mailroom_oo.py

### PR DESCRIPTION
**Pull request by tammytdo**

### CODE REVIEW

File: students/tammyd/lesson09/mailroom_oo.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    #pt1: a history of the amounts they have donated.
+    @property
+    def total_donations(self):
+        return sum(self._donations)
+
+    @property
+    def new_donation(self, donation):
+        self._donations.append(donation)
+
+    @property
```
[the remark this answers is not in the document]
Comment (PythonCHB) [reply students/tammyd/lesson09/mailroom_oo.py]: actually, this on is OK -- total_donations as a property makes sense, then you can do:

print("the total donated by this donor is:", donor.total_donations)

-CHB

File: students/tammyd/lesson09/mailroom_oo.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    def avg_donations(self):
+        return sum(self._donations) / len(self._donations)
+
+
+
+
+#You’ll then want a class that handles the collection of donors. 
+#method needs the information from the whole collection.
+#pt1: data structure that holds a list of your donors
+#Do I want to include or exclude () right after DonorDB here?
```
Comment (PythonCHB) [inline students/tammyd/lesson09/mailroom_oo.py]: no need for parens here -- you use parens when you subclass from something.

File: students/tammyd/lesson09/mailroom_oo.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+class DonorDB:
```
Comment (PythonCHB) [inline students/tammyd/lesson09/mailroom_oo.py]: fine start -- but you need a lot more here!

File: students/tammyd/lesson09/mailroom_oo.py [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+        # else:
+        #     break
+        except KeyError:
+            print("Incorrect entry. Try again.")
+            return
+    donation(donor_name)
+
+
+
+def find_donor(donor_name):
```
Comment (PythonCHB) [inline students/tammyd/lesson09/mailroom_oo.py]: this should be a method of your DonorDB class

File: students/tammyd/lesson09/mailroom_oo.py [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+    for donor in donor_db:
+        if donor_name.strip().lower() == donor[0].lower():
+            return donor
+    return None
+
+def ty_letter(donor_name, donor_amount):
```
Comment (PythonCHB) [inline students/tammyd/lesson09/mailroom_oo.py]: this should be a method of the Donor class.

File: students/tammyd/lesson09/mailroom_oo.py [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+    print("\nMenu: Send a Thank You, Letter\n")
+    letter = (f"\nDear {donor_name.title()},\n\n\tYour donation of ${donor_amount} will make a great difference to a young learner at the coding academy. Thank you.\n\n\nSincerely,\nAn Academy Student\n")
+    print(letter)
+    print("--------------------------------\n\n")
+    main_menu()
+
+
+def letters_to_everyone():
+    for donor_name in donor_db.values():
+        letter = (f"\nDear {donor_name[0]},\n\n\tYour donation of ${donor_name[1][-1]} will make a great difference to a young learner at the coding academy. Thank you.\n\n\nSincerely,\nAn Academy Student\n")
```
Comment (PythonCHB) [inline students/tammyd/lesson09/mailroom_oo.py]: this is the same code as the previous function! -- you should call that from this function.

File: students/tammyd/lesson09/mailroom_oo.py [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+
+def ty_letter(donor_name, donor_amount):
+    print("\nMenu: Send a Thank You, Letter\n")
+    letter = (f"\nDear {donor_name.title()},\n\n\tYour donation of ${donor_amount} will make a great difference to a young learner at the coding academy. Thank you.\n\n\nSincerely,\nAn Academy Student\n")
+    print(letter)
+    print("--------------------------------\n\n")
+    main_menu()
+
+
+def letters_to_everyone():
```
Comment (PythonCHB) [inline students/tammyd/lesson09/mailroom_oo.py]: method of DonorDB class...

are you getting the idea?

**Comment by tammytdo** [comment]
Hello Chris,

Thank you immensely. I am committed to re-studying all the material and
comments, and getting extra support before the Advanced course begins.

With gratitude,
Tammy

**Comment by PythonCHB** [comment]
I"m merging this, and giving you credit for OO mailroom, to get yu over the 80% ,mark, so you can move on to the next class.

But you really need to get basic OO down at some point -- it's only going to get harder next quarter, so I encourage you to work through both the HTML render and OO mailroom exercises -- if yo8u can get those cleanly done, you'll be in much better shape for moving through the rest of the program.

File: students/tammyd/lesson09/mailroom_oo.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hosungs) [reply students/tammyd/lesson09/mailroom_oo.py]: Oh, yes. Sorry for misleading. I must have gone through this PR too fast in my morning commute bus... Yes, this is OK, and actually this is how I did it in class...

**Pull request merged**